src/main/java/com/dubture/jenkins/digitalocean/SlaveTemplate.java

```diff
@@ -155,17 +157,19 @@
      * @param tags the droplet tags
      * @param userData user data for DigitalOcean to apply when building the slave
      * @param initScript setup script to configure the slave
+     * @param imageByName whether to resolve the image by name rather than id
      */
     @DataBoundConstructor
     public SlaveTemplate(String name, String imageId, String sizeId, String regionId, String username, String workspacePath,
                          Integer sshPort, Boolean setupPrivateNetworking, String idleTerminationInMinutes, String numExecutors, String labelString,
                          Boolean labellessJobsAllowed, String instanceCap, Boolean installMonitoring, String tags,
-                         String userData, String initScript) {
+                         String userData, String initScript, Boolean imageByName) {
```

Comment:
So I'm doing a quick pass on my phone. Is this actually used in the constructor or just setting?

I think changing the constructor can break loading older configs. I'll have to check. I think in general a @DataBoundSetter is safer. It might only matter if your removing fields.

I'll do a trial run with existing config and jcasc.

Reply:
I hadn't given it much thought TBH. Having a constructor that accepts all but one member feels a bit iffy though, so does having imageByName not be final when imageId is. That said, I'm not a fan of ctors with a million arguments of none-descriptive type anyway ^^

So, I drop the ctor argument and replace it with `setImageByName(Boolean imageByName)`?